### PR TITLE
Refactor "Routing/Builder" methods pre conditions

### PR DESCRIPTION
**Pull request by mottaghipour**
# Refactor "Routing/Builder" methods pre conditions

Replace block conditions with "ArgumentNullException.ThrowIfNull(...);"

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=41783) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=41783) before we can accept your contribution.<br/><br/>:x: mottaghipour [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=41783)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=41783) it.</sub>

File: src/Http/Routing/src/Builder/RoutingEndpointConventionBuilderExtensions.cs [hunk 1]
```diff
@@ -97,15 +80,8 @@ public static class RoutingEndpointConventionBuilderExtensions
...
-        if (builder == null)
-        {
-            throw new ArgumentNullException(nameof(builder));
-        }
-
-        if (items == null)
-        {
-            throw new ArgumentNullException(nameof(items));
-        }
+        ArgumentNullException.ThrowIfNull(builder, nameof(builder));
```
Comment (davidfowl) [inline src/Http/Routing/src/Builder/RoutingEndpointConventionBuilderExtensions.cs:83]: Why is nameof needed? Isn't it inferred by default from the expression passed in?

Comment (mottaghipour) [reply src/Http/Routing/src/Builder/RoutingEndpointConventionBuilderExtensions.cs:83]: Exactly. You are right and I was wrong.
@halter73 Had added MapGroup(https://github.com/dotnet/aspnetcore/pull/41265) pre conditions with param names and I continued the same.

Comment (mottaghipour) [reply src/Http/Routing/src/Builder/RoutingEndpointConventionBuilderExtensions.cs]: It just got fixed

**Review by javiercn — APPROVED** [review]

**Comment by mottaghipour** [comment]
Done

**Comment by halter73** [comment]
@mottaghipour Why did you close this PR. s there some reason you do not want us to merge this change?

**Comment by mottaghipour** [comment]
@halter73 After pulling, I read the [contributing guidelines](https://github.com/dotnet/aspnetcore/blob/ce38917e8ad8b0244c52b318ac25d7471dad77ec/CONTRIBUTING.md) and in this [part](https://github.com/dotnet/aspnetcore/blob/ce38917e8ad8b0244c52b318ac25d7471dad77ec/CONTRIBUTING.md#before-submitting-the-pull-request), I realized that my work was wrong, refactoring here is bad for merge!
But if you do not have a problem, merge this change. I am also happy for my contribution :)

**Comment by halter73** [comment]
Generally, we like people to file an issue first. Mostly it's so people don't waste time on PRs that we won't merge. In this particular case it's fine though. Thanks for your contribution!

**Pull request merged**